File: packages/go/sh/whoa/sutradhar/sdk/v1/upgrade_parity_test.go
package v1

import "testing"

func TestUpgradeParityRuleIDStability(t *testing.T) {
	e := BuildValidationError("common.idempotency.scope.required", "idempotency.scope", "ERROR")
	if e.RuleID != "common.idempotency.scope.required" {
		t.Fatalf("unexpected rule id: %s", e.RuleID)
	}
	if e.Code != "SH_WHOA_SUTRADHAR.common.idempotency.scope.required" {
		t.Fatalf("unexpected code: %s", e.Code)
	}
}

File: packages/typescript/src/sdk/upgrade-parity.test.mjs
import test from "node:test";
import assert from "node:assert/strict";
import { buildValidationError } from "./validation.mjs";

test("rule IDs remain stable for compatibility-sensitive checks", () => {
  const e = buildValidationError("common.idempotency.scope.required", "idempotency.scope", "ERROR");
  assert.equal(e.ruleId, "common.idempotency.scope.required");
  assert.equal(e.code, "SH_WHOA_SUTRADHAR.common.idempotency.scope.required");
});
